### PR TITLE
do not propagate events through portal

### LINKED ISSUE #1810: Focus/clicking issues with dialogs triggered from a table cell
# 🐛 Bug Report 

If a dialog is triggered from an action button in a table cell, clicking the dialog or the underlay will select the row from where the dialog was triggered. This prevents any dialog text selection or focus of any focusable children in the dialog content area. Clicking dialog buttons does still work however.

## 🤔 Expected Behavior

Clicks should not pass through the dialog/underlay and focusable children in the dialog should be focusable.

## 😯 Current Behavior

Clicks pass through the dialog (and select the table row). Focusable children in the dialog cannot be focused (e.g., textfields).

## 💻 Code Sample

https://codesandbox.io/s/blue-dew-z0woi?file=/src/App.js

Click any of the action buttons to trigger a dialog. Click the dialog body or underlay and observe that the row will toggle selection or that clicking the textfield will not focus the textfield for text input.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.9.0
| Browser          | Chrome
| Operating System | MacOS Catalina

## 🧢 Your Company/Team

Adobe/Quarry

### PR DESCRIPTION
**Pull request by jluyau**
Closes #1810 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

Adobe/Quarry

### CODE REVIEW

File: packages/@react-aria/interactions/test/useHover.test.js [hunk 1]
```diff
@@ -369,4 +374,86 @@ describe('useHover', function () {
...
+
+        fireEvent.click(res.getByText('open'));
+
+        act(() => {
+          jest.runAllTimers();
+        });
+
+        let el = res.getByText('test');
+        fireEvent.mouseEnter(el);
+        fireEvent.mouseLeave(el);
```
Comment (snowystinger) [inline packages/@react-aria/interactions/test/useHover.test.js:454]:
```suggestion
        start(el);
        end(el);
```

File: packages/@react-aria/interactions/test/useHover.test.js [hunk 2]
```diff
@@ -369,4 +374,86 @@ describe('useHover', function () {
...
+    });
+
+    describe.each`
+      type                | prepare               | actions
+      ${'Mouse Events'}   | ${installMouseEvent}  | ${[
+        (el) => fireEvent.mouseEnter(el, {button: 0}),
+        (el) => fireEvent.mouseLeave(el, {button: 0})
+      ]}
+      ${'Pointer Events'} | ${installPointerEvent}| ${[
+        (el) => fireEvent(el, pointerEvent('pointerover', {pointerType: 'touch'})),
```
Comment (snowystinger) [inline packages/@react-aria/interactions/test/useHover.test.js:432]: why touch? for the most part, i thought touch couldn't start hover, that's why there's not call to triggerHover* in any touch handlers in useHover

File: packages/@react-aria/interactions/test/usePress.test.js [hunk 3]
```diff
@@ -2033,4 +2038,93 @@ describe('usePress', function () {
...
+
+        fireEvent.click(res.getByText('open'));
+
+        act(() => {
+          jest.runAllTimers();
+        });
+
+        let el = res.getByText('test');
+        start(el, {});
+        end(el, {});
```
Comment (snowystinger) [inline packages/@react-aria/interactions/test/usePress.test.js:2125]: these functions only take one argument

**Review by snowystinger — COMMENTED** [review]
Out of curiosity, how did you decide or figure out which handlers needed this check?
for instance, why pointerup, but not pointermove?

**Comment by jluyau** [comment]
I wasn't sure if the other events (that I didn't modify) were being triggered in my manual tests. Didn't want to just add the check for every event if it wasn't triggering at all, but I can add to others if needed?

**Comment by snowystinger** [comment]
Agreed, there are a lot of different interaction paths, however, I think we need to really carefully address all of them. That means doing things like testing the screen reader in FF on windows or whatever else we can think of. Possibly talk to @devongovett  about the weirder ones and look through his blog post to track them down as well.

File: packages/@react-aria/interactions/src/usePress.ts [hunk 4]
```diff
@@ -214,7 +214,6 @@ export function usePress(props: PressHookProps): PressResult {
           e.preventDefault();
```
Comment (devongovett) [inline packages/@react-aria/interactions/src/usePress.ts:214]: Do we need to do the check for keyboard events as well?

File: packages/@react-aria/interactions/src/usePress.ts [hunk 5]
```diff
@@ -429,6 +440,10 @@ export function usePress(props: PressHookProps): PressResult {
       };
 
       pressProps.onMouseUp = (e) => {
+        if (!e.currentTarget.contains(e.target as HTMLElement)) {
+          return;
+        }
+
         if (!state.ignoreEmulatedMouseEvents && e.button === 0) {
           triggerPressUp(e, state.pointerType);
         }
```
Comment (devongovett) [inline packages/@react-aria/interactions/src/usePress.ts:449]: Make sure we get all of the events assigned to pressProps. I think maybe touch events are missing? Is there a reason they were not included?

**Review by devongovett — CHANGES_REQUESTED** [review]

File: packages/@react-aria/interactions/src/usePress.ts (hunk 4 above)
Comment (jluyau) [reply packages/@react-aria/interactions/src/usePress.ts:214]: yeah most likely. i'll see if i can add a good test for that

File: packages/@react-aria/interactions/src/usePress.ts (hunk 5 above)
Comment (jluyau) [reply packages/@react-aria/interactions/src/usePress.ts:449]: i think i had placed them all the functions with an event, but i couldn't tell if they were doing anything or not. should i go ahead and place them in all the pressProps events?

Comment (devongovett) [reply packages/@react-aria/interactions/src/usePress.ts:449]: Yeah all of the ones that are on pressProps, not global listeners.

**Review by devongovett — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**